integration-test/src/main/java/com/sequenceiq/it/cloudbreak/assertion/audit/AuditGrpcServiceAssertion.java

```diff
@@ -178,4 +182,29 @@ protected boolean shouldCheckFlowEvents() {
     }
 
     protected abstract Crn.Service getService();
+
+    private void retry(OperationInfo operationInfo, Runnable runnable) {
+        int maxRetry = 3;
+        int attempt = 0;
+        long waitingSec = 1;
+        boolean running = true;
+        while (running && attempt < maxRetry) {
+            try {
+                runnable.run();
+                running = false;
+            } catch (Exception e) {
+                attempt++;
+                LOGGER.info("Cannot validate the audit logs: {}, try the next attempt", operationInfo.getEventName(), e);
```

Comment:
Exception `e` will never be logged.
Please use String.format for the first argument.

Reply:
what is the reason for using the String.format instead of the logger format?

Reply:
please check your signature in IDEA. (or slf4j docs). Your usage maps to `public void info(String format, Object arg1, Object arg2);`
The signature that correctly handles Throwable is `public void info(String msg, Throwable t);`

Reply:
I've thought correctly, works like a charm. 
I executed this: 
```
LOGGER.info("Cannot validate the audit logs: {}, try the next attempt", operationInfo.getEventName(), new Exception("This is an example"));
```

This is the output:
```
11:52:03.859 [main] INFO  c.s.i.c.a.a.AuditGrpcServiceAssertion [LdapConfigTest.testCreateValidLdap] - Cannot validate the audit logs: CreateLdapConfig, try the next attempt
java.lang.Exception: This is an example
	at com.sequenceiq.it.cloudbreak.assertion.audit.AuditGrpcServiceAssertion.retry(AuditGrpcServiceAssertion.java:191)
	at com.sequenceiq.it.cloudbreak.assertion.audit.AuditGrpcServiceAssertion.executeAuditValidation(AuditGrpcServiceAssertion.java:64)
	at com.sequenceiq.it.cloudbreak.assertion.audit.AuditGrpcServiceAssertion.create(AuditGrpcServiceAssertion.java:35)
	at com.sequenceiq.it.cloudbreak.context.TestContext.then(TestContext.java:366)
	at com.sequenceiq.it.cloudbreak.dto.AbstractFreeIpaTestDto.then(AbstractFreeIpaTestDto.java:92)
	at com.sequenceiq.it.cloudbreak.dto.AbstractFreeIpaTestDto.then(AbstractFreeIpaTestDto.java:87)
	at com.sequenceiq.it.cloudbreak.testcase.mock.LdapConfigTest.testCreateValidLdap(LdapConfigTest.java:63)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.MethodInvocationHelper$1.runTestMethod(MethodInvocationHelper.java:238)
	at org.springframework.test.context.testng.AbstractTestNGSpringContextTests.run(AbstractTestNGSpringContextTests.java:181)
	at org.testng.internal.MethodInvocationHelper.invokeHookable(MethodInvocationHelper.java:252)
	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:595)
	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:174)
	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:822)
	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:147)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.testng.TestRunner.privateRun(TestRunner.java:764)
	at org.testng.TestRunner.run(TestRunner.java:585)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:384)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:378)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:337)
	at org.testng.SuiteRunner.run(SuiteRunner.java:286)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1218)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
	at org.testng.TestNG.runSuites(TestNG.java:1069)
	at org.testng.TestNG.run(TestNG.java:1037)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)
```

Reply:
Accepted, I have also checked with a sample code.